### PR TITLE
feat(kit): add `isNuxtMajorVersion` compatibility util, deprecate `isNuxt2` and `isNuxt3`

### PR DESCRIPTION
**Pull request by lihbr**
### 🔗 Linked issue

/

### 📚 Description

This PR adds a `isNuxt4` utils that works like the existing `isNuxt2` and `isNuxt3` ones to `@nuxt/kit`.

Maybe a bit early to add it but I had to check for it preemptively already on a module so maybe that's what we want?

I've tried to check if this was done somewhere else already but couldn't, sorry if this is duplicate 🙏

### CODE REVIEW

**Comment by DamianGlowala** [comment]
I'd suggest introducing a single util instead for any future Nuxt versions moving forward:
```
export function isNuxtVersion (majorVersion: number, nuxt: Nuxt = useNuxt()) {
  const version = getNuxtVersion(nuxt)
  return version.startsWith(`${majorVersion}.`)
}
```

**Comment by lihbr** [comment]
I like that also! Happy to refactor in that direction if we agree on that

**Review by danielroe — APPROVED** [review]

**Comment by danielroe** [comment]
Thank you for this - and yes, I like @DamianGlowala's idea too!

**Comment by lihbr** [comment]
Will refactor in the evening then ☺️ hold on!

**Comment by lihbr** [comment]
Done:
1. I know we've been avoiding `startsWith` when possible for performance reasons, I believe it's not applicable here as it could break the logic when we reach Nuxt 10 (well, not happening tomorrow ^^'). Maybe we want to compromise here and keep using `version[0]`?
2. `@nuxt/kit` appears with a leading `\` in the deprecation notice as otherwise it's considered as a TSDoc tag an rendered differently. The escape character doesn't show as expected when hovering the definition 👌

File: packages/kit/src/compatibility.ts [hunk 1]
```diff
@@ -98,19 +98,26 @@ export async function hasNuxtCompatibility (constraints: NuxtCompatibility, nuxt
...
 /**
- * Check if current nuxt instance is version 2 legacy
+ * Check if current Nuxt instance is of specified major version
  */
-export function isNuxt2 (nuxt: Nuxt = useNuxt()) {
+export function isNuxtMajorVersion (majorVersion: 2 | 3 | 4, nuxt: Nuxt = useNuxt()) {
   const version = getNuxtVersion(nuxt)
-  return version[0] === '2' && version[1] === '.'
+
+  return version.startsWith(`${majorVersion}.`)
```
Comment (DamianGlowala) [inline packages/kit/src/compatibility.ts:106]: Following your advice, let's revert it to a more performance-oriented version! 😄 
```suggestion
  return version[0] === majorVersion.toString() && version[1] === '.'
```

**Review by DamianGlowala — CHANGES_REQUESTED** [review]

**Review by danielroe — APPROVED** [review]

**Review by DamianGlowala — APPROVED** [review]

**Pull request merged**